types/index.d.ts

```diff
@@ -4,7 +4,7 @@ declare module 'react-easy-state' {
   // takes an object (optionally), wraps it in a transparent proxy and returns the proxy
   function store<Store extends object>(obj?: Store): Store;
   // takes class or function component and returns a class HOC
-  function view<Comp extends ComponentType<any>>(comp: Comp): Comp;
+  function view<T extends React.ComponentType<any>>(comp: T): T;
```

Comment:
Out of curiosity: is this any different than the original line?

Reply:
Nope sorry, I played around with this part a lot, but after reverting it ended up looking like this.

Any help would be appreciated with the following problem: I wanted to get rid of the any from the function by using something more specific, but it looks like in this case only any works and TypeScript is able to figure out the correct types. (which is weird)

Basic definition would be
```ts
function view<T>(comp: T): T
```
If we want to specify that T is a react component:
```ts
function view<T extends React.ComponentType>(comp: T): T
```
But that causes a problem: `React.ComponentType` is generic (has a Type parameter P) which are the props. If we were to say something like the above it won't work bc. the type parameter will be `{}` which means it gets injected into `ReactProps<{}>` which is an alias for `{} & { children, ref }` which is just `{ children, ref }`

So I thought let's add:
```ts
function view<P extends React.Prop,T extends React.ComponentType<P>>(comp: T<P>): T<P>
```

And that didn't work and I feel kind of stuck. Why?
`<any>` shouldn't work in our case but TS compiler is able to infer the type of any for some reason, from the following.

```typescript
type AppProps = {
  greeting: string;
}

const App<AppProps> = view(({ greeting }) => {
  return <div>Hello</div>
});
```

Which is amazing but I just want to know why, and I want to add to the documentation how to annotate in order for this to work, assigning the view to a `const` works so far which probably covers 90% of the use-cases, but I think we should investigate a bit more what other use-cases can be that would affect this.

Reply:
I tried to look into this but I have no idea at all.

>>
So I thought let's add:
function view<P extends React.Prop,T extends React.ComponentType<P>>(comp: T<P>): T<P>
And that didn't work and I feel kind of stuck. Why?

What was the error?

Reply:
`Property 'greeting' does not exist on type '{ children?: ReactNode; }'.  TS2339` Looks like it did not infer the types properly as it did with any.

neither with
```typescript
  function view<P extends {}, T extends React.ComponentType>(
    comp: T<P>,
  ): T<P>;
```

nor with
```typescript
  function view<P extends {}, T extends React.ComponentType<P>>(
    comp: T,
  ): T;
```